napari/layers/mixins.py

```diff
@@ -0,0 +1,148 @@
+import warnings
+
+import numpy as np
+from vispy.color import Colormap
+
+from ..utils.colormaps import make_colorbar
+from ..utils.event import Event
+from ..utils.status_messages import format_float
+from ..utils.validators import validate_n_seq
+from .layer_utils import increment_unnamed_colormap
+
+
+validate_2_tuple = validate_n_seq(2)
+
+
+class ImageSurfaceMixin:
```

Comment:
Should we name this whole file `image_surface_mixin.py` instead of just `mixins.py`. I feel if we add more mixins might be nicer to keep them separate rather than all in here